myJoin.pl

```diff
@@ -98,33 +103,28 @@ sub line_hash1{
         @headers = split("\t", $header_text);
         %header_lib = map {$headers[$_] => $_} 0..$#headers;
     }
-#    print "reading $input\n";
+#     print "headers: @headers\n";
+# 	  print Dumper %header_lib;
     while (<FILE>){
+    	chomp;
         @row = split("\t");
-#        map {print "my row $_\n";} @row;
         my $key = '';
-        foreach my $k (@field){
+        foreach my $k (@fields){
             $ref = $header_lib{$k};
             if (exists $row[$ref]){
-            $key .=  $row[$ref];
-#           map {print "my row: $_\n";} @row;
+            	$key .=  $row[$ref];
             }else{
                 die "input file contains empty fields";
-                }
-#           print "my key: $key\n";
-
-           
-            }
-        #This field
-        foreach my $h (keys %header_lib){
-            foreach my $v (@field){
-            if ($h ne $v){
-            my $i = $header_lib{$h};
-            $attribu{$key}{$h} = $row[$i];
-                }
             }
         }
-
+        print "my key: $key\n";
+    
+        # Add non-key fields to attribu dictionary
+        foreach my $h (keys %header_lib){
+            next if( defined $field_names{$h} );
+			my $i = $header_lib{$h};
+			$attribu{$key}{$h} = $row[$i];
+		}
```

Comment:
This is the new code for adding values of non-key fields to the `attribu` dictionary. The outer loop still iterates over each field in the current file, but we remove the inner loop, simply checking if each field is a key field by definition in the [`%field_names` hash](https://github.com/LMT337/myJoin/pull/1/files#diff-bf1c953fe5d72a0f9f7f2732f7e61d0fR40).